### ISSUE TITLE
Feature request -> Intellij IDE plugin for JSX

### ISSUE THREAD

**Issue by donjaime** [jsx/JSX#255]
Been using JSX for about a year for our project, and it has been great so far. But IDE support for things like TypeScript have made it increasingly appealing to switch in orderer to keep high developer productivity as the codebase gets larger.

Was curious if anyone had plans to deliver proper IDE support in something like IntelliJ. JSX's static typing should make very amenable to really spectacular IDE support, which might arguably have a much bigger impact on productivity than continued language changes or compiler optimizations. 

I realize that the team is busy and might not have available bandwidth to work on such a thing. But I figured I would ask anyways.

http://confluence.jetbrains.com/display/IDEADEV/Developing+Custom+Language+Plugins+for+IntelliJ+IDEA

Thanks!

**Comment by bpatters** [comment]
+100. Having to switch between web storm and intellij idea is a pain in the butt for us full stack developers.

**Comment by moduscreate** [comment]
Maybe this should be done by Jetbrains?

**Comment by davidrenne** [comment]
Yes it should be done by jetbrains, but also there are open source plugins that others write and when you open up a JSX file it allows you to download.

So either way.  But a +1 for sure.  So far at least the JSX files show no parse errors when I map jsx as javascript files.

**Comment by ticofab** [comment]
+1

**Comment by gnashe** [comment]
+1

**Comment by bZichett** [comment]
+1

**Comment by forceoflight** [comment]
+1

**Comment by bZichett** [comment]
+1

**Comment by Rayraegah** [comment]
I build a lot of in house tools to support our use of JSX in sublime text IDE since early 2015; like syntax highlight, theme support for JSX specific keywords, esFormat (beautify), auto-complete snippets, hints and build tools in python.

Here's the [Sublime Text language file](https://github.com/Rayraegah/jsx-sublime) if anyone's interested.

EDIT: [Typescript](https://github.com/Microsoft/TypeScript-Sublime-Plugin) syntax definitions can also be used
